CRM/Contact/Page/DedupeFind.php

```diff
@@ -53,10 +70,18 @@ public function getBAOName() {
   public function &links() {
   }
 
+  /**
+   * Initialize properties from input.
+   */
+  protected function initialize() {
+    $this->selected = CRM_Utils_Request::retrieveValue('selected', 'Boolean');
+  }
+
   /**
    * Browse all rule groups.
    */
   public function run() {
+    $this->initialize();
     $gid = CRM_Utils_Request::retrieve('gid', 'Positive', $this, FALSE, 0);
```

Comment:
Really I'd like to move all the retrieve from url functions into initialize but I'd like to have the AlphaNumeric rule merged first